handsontable/src/validators/autocompleteValidator/autocompleteValidator.js

```diff
@@ -47,7 +47,7 @@ function process(value, callback) {
     let found = false;
 
     for (let s = 0, slen = source.length; s < slen; s++) {
-      if (originalVal === source[s]) {
+      if (JSON.stringify(originalVal) === JSON.stringify((source[s]))) {
```

Comment:
The [`structuredClone`](https://developer.mozilla.org/en-US/docs/Web/API/Window/structuredClone) API has good browser coverage.
```suggestion
      if (structuredClone(originalVal) === structuredClone((source[s]))) {
```